src/plugins/explore/public/application/register_tabs.ts

```diff
@@ -45,17 +47,31 @@ export const registerBuiltInTabs = (tabRegistry: TabRegistryService) => {
     supportedLanguages: [EXPLORE_DEFAULT_LANGUAGE],
 
     prepareQuery: (query) => {
-      return typeof query.query === 'string' ? query.query : '';
+      const state = services.store.getState();
+
+      // Get the selected patterns field from the Redux state
+      let patternsField = state.tab?.patterns?.patterns_field;
```

Comment:
```suggestion
      let patternsField = state.tab.patterns.patterns_field;
```

Reply:
I intentionally included these because I believe there were situations where I've seen patterns_field not be a field inside of options? Although should I be making sure this is never the case?

Reply:
changed anyways